app/controllers/assignments_controller.rb

```diff
@@ -832,7 +832,22 @@ def process_assignment_form(assignment)
       periods = period_attrs.to_h.values.map { |h| h[:id].presence }
       assignment.submission_rule.periods.where.not(id: periods).find_each(&:destroy)
     end
-    assignment.assign_attributes(assignment_params)
+
+    # Handle "scheduled" visibility option
+    # When is_hidden="scheduled", convert to is_hidden=false with datetime values
+    params_copy = assignment_params.deep_dup
+    if params_copy[:is_hidden] == 'scheduled'
```

Comment:
I'm wary of having the string literal `'scheduled'` shared in multiple places.

You can define a constant (really in `Assessment.rb`) and use it both here and in the front-end (note that you can use embedded Ruby in the `.erb` file).